mcp/src/main/java/io/airlift/mcp/reference/Mapper.java

```diff
@@ -85,7 +89,24 @@ static McpSchema.Role mapRole(Role ourRole)
         return McpSchema.Role.valueOf(ourRole.name().toUpperCase());
     }
 
-    static McpStatelessServerFeatures.SyncResourceSpecification mapResource(Resource ourResource, ResourceHandler ourHandler)
+    static McpRequestContext buildMcpRequestContext(McpTransportContext mcpTransportContext, Optional<Map<String, Object>> meta, RequestContextProvider requestContextProvider)
+    {
+        Optional<Object> progressToken = meta.flatMap(m -> {
```

Comment:
Can we represent token as an class/record or a sealed interface?

```
record ProgressToken(Object value)
{
}
```

or 

```
    sealed interface ProgressToken
    {
        record StringProgressToken(String token) implements ProgressToken
        {
        }

        record LongProgressToken(long token) implements ProgressToken
        {
        }
    }
```

`Object` tells nothing